actionmailer/test/delivery_methods_test.rb

```diff
@@ -102,16 +102,21 @@ def welcome(hash={})
   end
 
   test "ActionMailer should be told when Mail gets delivered" do
-    DeliveryMailer.expects(:deliver_mail).once
-    DeliveryMailer.welcome.deliver_now
+    DeliveryMailer.delivery_method = :test
```

Comment:
I was worried that changing the `delivery_method` would spill over into other tests, but the setup and teardown makes sure to keep the old delivery method: https://github.com/rails/rails/pull/21306/files#diff-db5e616c1ed1b103bd61d787ad6cf567R100

So there's no problem.